server/sched.go

```diff
@@ -215,6 +215,8 @@ func (s *Scheduler) processPending(ctx context.Context) {
 						g := pickBestFitGPUs(pending, ggml, gpus, &numParallel)
```

Comment:
Does `pickBestFitGPUs` return `nil` if there are multiple candidates available? It wasn't super obvious why `pickBestPartialByLibrary` was only called then – perhaps we should sort and take the one with the most VRAM?

Reply:
I'll improve the comments to explain it better.

`pickBestFitGPUs` already is [brand aware](https://github.com/ollama/ollama/blob/main/server/sched.go#L686), and only returns a list of GPUs if the model completely fits into memory.  If it can't find a single brand collection of GPUs that fully fit, it returns nil.  This informs the scheduler that we're in a scenario where some other model has to be unloaded to make room, or this is the first model load.  It's that last case which was the bug this PR is fixing.  For first model load, if the model doesn't fully fit into one brand collection of GPUs, then we'd fall through to a chunk of code that wound up passing the whole list of GPUs with mixed brands into the prediction and server startup logic.